src/user.rs

```diff
@@ -182,28 +145,31 @@ impl User {
             ) {
                 Ok(result)
             } else {
-                Err(JsonSaveError::WriteFile)
+                Err(Error::WriteFile)
             }
         } else {
-            Err(JsonSaveError::Serialize)
+            Err(Error::Serialize)
         }
     }
 
-    pub async fn load(id: &ID) -> Result<Self, JsonLoadError> {
-        if let Ok(json) =
-            tokio::fs::read_to_string(USER_FOLDER.to_owned() + &id.to_string() + ".json").await
-        {
+    pub async fn load(id: &ID) -> Result<Self> {
+        let filename = USER_FOLDER.to_owned() + &id.to_string() + ".json";
+        let path = std::path::Path::new(&filename);
+        if !path.exists() {
+            return Err(Error::HubNotFound);
+        }
+        if let Ok(json) = tokio::fs::read_to_string(path).await {
             if let Ok(result) = serde_json::from_str(&json) {
                 Ok(result)
             } else {
-                Err(JsonLoadError::Deserialize)
+                Err(Error::Deserialize)
             }
         } else {
-            Err(JsonLoadError::ReadFile)
+            Err(Error::ReadFile)
         }
```

Comment:
mix and mash `map_err` with `map` seems a good idea. Keep them if you want, as always.